azext_iot/central/models/__init__.py

```diff
@@ -3,3 +3,92 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+from abc import abstractmethod, ABCMeta
+
+
+class BaseTemplate(metaclass=ABCMeta):
+    def __init__(self, template: dict):
+        self.id = None
+        self.schema_names = None
+        self.raw_template = template
+        try:
+            self.name = template.get("displayName")
+            self.components = self._extract_components(template)
+            if self.components:
+                self.component_schema_names = self._extract_schema_names(
+                    self.components
+                )
+
+        except Exception:
+            raise CLIError("Could not parse iot central device template.")
+
+    def _get_schema_name(self, schema) -> str:
+        return "name" if "name" in schema else "@id"
+
+    def _extract_schemas(self, entity: dict) -> dict:
+        schema = entity.get("schema")
+        if schema is not None:
```

Comment:
this can be simplified to an if, elif, else

Reply:
+1